include/dlaf/matrix/matrix_const.tpp

```diff
@@ -31,25 +31,28 @@ Matrix<const T, device>::Matrix(Distribution distribution, const matrix::LayoutI
   setUpTiles(mem, layout);
 }
 
-template <class T, Device device>
-Matrix<const T, device>::~Matrix() {
-  for (auto&& tile_manager : tile_managers_) {
-    try {
-      tile_manager.clearSync();
-    }
-    catch (...) {
-      // TODO WARNING
-    }
-  }
-}
-
 template <class T, Device device>
 hpx::shared_future<Tile<const T, device>> Matrix<const T, device>::read(
     const LocalTileIndex& index) noexcept {
-  std::size_t i = to_sizet(tileLinearIndex(index));
+  const auto i = tileLinearIndex(index);
   return tile_managers_[i].getReadTileSharedFuture();
 }
 
+void Matrix<const T, device>::syncAll() noexcept {
```

Comment:
I don't know if the name of this barrier function is the best.

My first idea is it should just synchronize local tiles and just care about them, leaving flexibility to the user to decide to create a more "global" barrier involving other ranks (e.g. via an `MPI_Barrier`).

Do you like this kind of API? Should it be something more `syncLocal()`?

@msimberg @teonnik @rasolca

Reply:
How about `waitAllLocalTiles()`?